pkg/ldap/datasource.go

```diff
@@ -175,121 +401,22 @@ func (d *Datasource) GetPage(ctx context.Context, request *Request) (*Response,
 		return nil, validationErr
 	}
 
-	tlsConfig := &tls.Config{}
-
-	if request.IsLDAPS {
-		decodedCertChain, err := base64.StdEncoding.DecodeString(request.CertificateChain)
-		if err != nil {
-			return nil, &framework.Error{
-				Message: fmt.Sprintf("Failed to load certificates - %v", err),
-				Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INVALID_DATASOURCE_CONFIG,
-			}
-		}
-
-		caCertPool := x509.NewCertPool()
-		caCertPool.AppendCertsFromPEM(decodedCertChain)
-		tlsConfig.RootCAs = caCertPool
-		tlsConfig.ServerName = request.Host
-	}
-
-	ldapConn, err := ldap_v3.DialURL(request.BaseURL, ldap_v3.DialWithTLSConfig(tlsConfig))
-	if err != nil {
-		return nil, &framework.Error{
-			Message: fmt.Sprintf("Failed to dial ldap server - %v", err),
-			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INTERNAL,
-		}
-	}
-
-	// Bind credentials with connection
-	bindErr := ldapConn.Bind(request.BindDN, request.BindPassword)
-	if bindErr != nil {
-		return nil, &framework.Error{
-			Message: fmt.Sprintf("Failed to bind credentials - %v", bindErr),
-			Code:    api_adapter_v1.ErrorCode_ERROR_CODE_DATASOURCE_AUTHENTICATION_FAILED,
-		}
-	}
-	defer ldapConn.Close()
-
-	// Set cursor from the request (if exists)
-	pageControl := ldap_v3.NewControlPaging(uint32(request.PageSize))
-
-	if request.Cursor != nil && request.Cursor.Cursor != nil {
-		pageInfo, decodeErr := DecodePageInfo(request.Cursor.Cursor)
-		if decodeErr != nil {
-			return nil, decodeErr
-		}
-
-		cookie, err := OctetStringToBytes(*pageInfo.NextPageCursor)
-		if err != nil {
-			return nil, &framework.Error{
-				Message: fmt.Sprintf("Failed to parse cursor value - %v", err),
-				Code:    api_adapter_v1.ErrorCode_ERROR_CODE_INTERNAL,
-			}
-		}
-
-		pageControl.SetCookie(cookie)
+	ci, ok := connector.FromContext(ctx)
+	if ok {
+		return d.Client.ProxyRequest(ctx, &ci, request)
 	}
```

Comment:
nit: We only look to use `ci` here; let's limit the scope of vars where possible
```suggestion
	if ci, ok := connector.FromContext(ctx); ok {
		return d.Client.ProxyRequest(ctx, &ci, request)
	}
```

Reply:
I agree.